Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java

```diff
@@ -123,6 +123,7 @@ private AzureManager getAzureManager(AuthMethod authMethod) throws IOException {
                 if (localAzureManagerRef == null) {
                     switch (authMethod) {
                         case AD:
+                        case DC:
```

Comment:
Looks like we need a new `DcAuthManager` class which can inherit `AdAuthManager`

Reply:
AdAuthManager is a singleton class. Instead of inherit from it, I will create a base class `BaseADAuthManager` for `DCAuthManager` and `AdAuthManager` . The `AdAuthManager` is actually handling the interface login, but there are so many usages of this class, so I am not going to change it in this change list, just extract the common part to the base class. Then we could separate the `deviceLogin()` out.